### PR TITLE
Refactor testgrid file locations

### PR DESCRIPTION
**Pull request by fejta**
Create the following structure:
* `/testgrid`
  - `config.yaml` - description of test groups, tabs and dashboards.
  - `config.proto` - description of the proto config message we upload to GCS
  - `state.proto` - description of the proto grid messages we upload to GCS for each test group.
  - `/testgrid/cmd` - binaries
    - `/testgrid/cmd/config` - existing binary to read `config.yaml` to and write `config.pb`
    - `/testgrid/cmd/updater` - existing prototype to create `Grid` messages from a GCS folder
  - `/testgrid/yaml2proto` - existing library to take `config.yaml`, apply defaults and create a config proto message.
 

In other words from `config`:
* move `config.yaml`, `config.proto` to `/testgrid`
* move `config/*.go` to `/testgrid/cmd/config`
* move `yaml2proto` to `/testgrid/yaml2proto`
* Move the `:testgrid-config` `genrule` to `/testgrid/BUILD`

From `updater`:
* move `state.proto` to `/testgrid`
* move `updater/*.go` to `/testgrid/cmd/updater`

Next I intend to combine the `cmd/config` and `yaml2proto` packages into `cmd/config`. Will also probably rename the `cmd/{config,updater}` to something slightly different (they both are both updaters.. One is for config and the other testgroups)

/assign @krzyzacy @michelle192837

### CODE REVIEW

**Comment by michelle192837** [comment]
Whoops, thanks for catching XD
